float32[] x_coords
float32[] y_coords
